lib/api/core/models/repositories/repository.js

```diff
@@ -248,18 +238,6 @@ class Repository {
    * @returns {Promise}
    */
   delete (object, options = {}) {
-    if (! object) {
-      return Bluebird.reject(errorsManager.get(
```

Comment:
This module is only called by internal function so we should trust that its parameters are correct.